### ISSUE TITLE
refs are being created without specified in static.config

### ISSUE THREAD

**Issue by iuscare** [studiometa/js-toolkit#92]
Hey there,

the docs says that we should define the refs of the components by specifying their name in the configuration. I realized that those refs seem to exist even without specifying them in the config, as long as they have been set in the html markup.

Example:
```
<div data-component="Example">
   <div data-ref="foo">Foo</div>
</div>
```
```
import Base from "@studiometa/js-toolkit";

export default class Example extends Base {
  static config = {
    name: 'Example'
  };

  mounted() {
    console.log(this.$refs.foo); // works -> <div data-ref="foo">Foo</div>
  }
}

```

I am wondering if this is intended? I interpreted the docs to mean that one should explicitly define those refs, which should be created within the config object. If the current behavior is intended, could you please provide an example, for which use case it is then necessary to define the refs within the config object at all? 

Please don't get me wrong, I do not want to show a critical attitude, just want to ask to understand. 😊

**Comment by titouanmathis** [comment]
Hello @iuscare, 

This is by design 🙂 

In the first versions of this toolkit (alpha and beta), we did not need to define the refs used by the component as they were resolved by the selector `[data-ref]` inside the component and not inside a child component. 

```html
<div data-component="Parent">
  <!-- `this.$refs.btn` in the Parent class -->
  <button data-ref="btn">...</button>  
  <div data-component="Child">
    <!--`this.$refs.btn` in the `Child` class, not directly accessible from the `Parent` class -->
    <button data-ref="btn">...</button>
  </div>
```

While maintaining projects using complex components with multiple refs, it became hard to know when opening an old component which refs were available in the `this.$refs` object. So we introduced a `config.refs` property to list them all and facilitate the maintenance and evolution of existing components. 

But to avoid a big breaking change between the beta version and the 1.0.0 release, we kept the same resolution with the `[data-ref]` selector. A warning is added to the console to advise for a fully defined `config.refs` property.

You will find more information on the subject in #56.

Hope this helps!
